third_party/tsl/tsl/platform/cpu_info.cc

```diff
@@ -458,18 +464,50 @@ class CPUIDInfo {
     }
 #endif  // !PLATFORM_WINDOWS
   }
+  static void InitializeCPUFeature() {
+    // Initialize CPUIDInfo pointer.
+    if (cpuid != nullptr) return;
+
+    cpuid = new CPUIDInfo;
+    // Make sure CPUID registers are available before reading them.
+    if (!(getauxval(AT_HWCAP2) & ARM_COMPUTE_CPU_FEATURE_HWCAP2_BF16)) {
+      return;
+    }
```

Comment:
The code doesn't match the comment here. It doesn't "make sure CPUID registers are available". It just returns without setting `has_bf16`.

I also think the `if` is not necessary since lines 476-478 are doing the same thing: read again, do the bitwise and again, and also has `has_bf16 = 0` when the feature is not detected.